src/utils/manager.py

```diff
@@ -185,55 +186,128 @@ def mount(
         options: Mount options to pass when mounting NFS share.
 
     Raises:
-        Error: Raised if NFS share mount operation fails.
+        Error: Raised if the mount operation fails.
     """
-    # Convert NFS URL to `mount.nfs` format if endpoint is NFS URL.
+
+    def write_file(path: Union[str, os.PathLike], contents: str) -> None:
+        """Writes a new file in `path` with `contents`, truncating it if it exists.
+
+        Raises:
+            Error: Raised if the file could not be created or written.
+        """
+        try:
+            pathlib.Path(path).write_text(contents)
+        except OSError as e:
+            _logger.error(f"Failed to write file {path}. Reason:\n{e}")
+            raise Error(f"Failed to mount {mountpoint}")
+
+    # Convert `endpoint` to the `mount.nfs` format if `endpoint` is an NFS URL.
     if _nfs_url_check.match(endpoint):
         endpoint, port = _translate(endpoint)
     else:
         port = None
 
-    if not (target := pathlib.Path(mountpoint)).exists():
-        _logger.debug(f"Creating mountpoint {target}")
-        target.mkdir()
+    # Avoids TOCTOU
+    _logger.debug(f"Ensuring mountpoint at {mountpoint} exists")
+    pathlib.Path(mountpoint).mkdir(exist_ok=True)
+
+    autofs_id = _mountpoint_to_autofs_id(mountpoint)
 
-    cmd = ["mount", "-t", "nfs"]
+    mount_opts = ""
     if options:
         if port:
             options.append(f"port={port}")
-        cmd.extend(["-o", ",".join(options)])
-    cmd.extend([endpoint, target])
+        mount_opts = ",".join(options)
+
+    _logger.debug(f"Mounting NFS share endpoint {endpoint} at {mountpoint} with options {options}")
+
+    master = f"/- /etc/auto.{autofs_id}" + (f" {mount_opts}" if mount_opts else "")
+
+    write_file(f"/etc/auto.master.d/{autofs_id}.autofs", master)
+    write_file(f"/etc/auto.{autofs_id}", f"{mountpoint} {endpoint}")
```

Comment:
As with my previous comment, let's just use `.write_text` directly rather than using a wrapper as it's less complexity.

```suggestion
    pathlib.Path(f"/etc/auto.master.d/{autofs_id}.autofs").write_text(master)
    pathlib.Path(f"/etc/auto.{autofs_id}").write_text(f"{mountpoint} {endpoint}")
```